diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/NewlinesRule.kt

```diff
@@ -294,25 +330,42 @@ class NewlinesRule(private val configRules: List<RulesConfig>) : Rule("newlines"
     private fun ASTNode.isDotFromPackageOrImport() = elementType == DOT &&
             parent({ it.elementType == IMPORT_DIRECTIVE || it.elementType == PACKAGE_DIRECTIVE }, true) != null
 
-    /**
-     *  taking all expressions inside complex expression until we reach lambda arguments
-     */
     private fun ASTNode.isCallsChain() = getParentExpressions()
             .lastOrNull()
             ?.run {
                 mutableListOf<ASTNode>().also {
-                    getAllLeafsWithSpecificType(DOT, it)
-                    getAllLeafsWithSpecificType(SAFE_ACCESS, it)
+                    getOrderedCallExpressions(psi, it)
                 }
             }
+            ?.dropWhile { !it.treeParent.textContains('(') && !it.treeParent.textContains('{') }
+            // fixme: we can't distinguish fully qualified names from chain of calls for now
```

Comment:
```suggestion
            // fixme: we can't distinguish fully qualified names (like java.lang) from chain of property accesses (like list.size) for now
```
Just to be sure we don't forget what this was about